src/main/java/orm/dirty_check/DirtyCheckMarker.java

```
@@ -0,0 +1,56 @@
+package orm.dirty_check;
+
+import orm.TableEntity;
+import orm.TableField;
+import orm.exception.DirtyCheckNotAllowedException;
+
+import java.util.List;
+
+public class DirtyCheckMarker<E> {
+
+    private final TableEntity<E> entity;
+    private final TableEntity<E> snapshotEntity;
+
+    public DirtyCheckMarker(TableEntity<E> entity, TableEntity<E> snapshotEntity) {
+        this.entity = entity;
+        this.snapshotEntity = snapshotEntity;
+        throwIfNotSameEntity(entity, snapshotEntity);
+        throwIfIdNotSame(entity, snapshotEntity);
```

validation로직은 this로 할당되기전에 fail fast하게 처리해볼 수 있겠네요 :)